File: db_helpers.py
from pymongo import MongoClient


client = MongoClient("mongodb://localhost:27017/")
db = client["first-research"]


def persist_data_to_db(collection_name, data):
    collection = db[collection_name]
    try:
        data_id = collection.insert_one(data).inserted_id
        print("Successfully persisted data at ID: ", data_id)
    except Exception as e:
        print("Failed to persist data: ", e)


def get_all_data_from_db(collection_name):
    collection = db[collection_name]
    try:
        data = collection.find()
        print("Successfully fetched data")
        return data
    except:
        print("Failed to fetch data")


def update_data_in_db(collection_name, id, data):
    collection = db[collection_name]
    try:
        filter = {"_id": id}
        update = {"$set": data}
        collection.update_one(filter, update)
        print("Successfully updated data for ID: ", id)
    except:
        print("Failed to update")

File: refactorings_collection/src/main/java/com/research/DbHelpers.java
package com.research;

import com.mongodb.client.MongoClients;
import com.mongodb.client.FindIterable;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoDatabase;
import com.mongodb.client.MongoCollection;
import org.bson.Document;

public class DbHelpers {
    public static MongoDatabase database;

    public void establishDBConnection() {
        try {

            MongoClient mongoClient = MongoClients.create("mongodb://localhost:27017");
            // Connect to the "mydb" database
            database = mongoClient.getDatabase("first-research");

        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public void persistToDB(String collectionName, Document data) {
        try {
            // Get a collection
            MongoCollection<Document> collection = database.getCollection(collectionName);

            collection.insertOne(data);
            System.out.println("Successfully persisted data");
        } catch (Exception e) {
            // TODO: handle exception
            System.out.println("Failed to persist data");
            System.out.println(e);
        }

    }

    public FindIterable<Document> getAllDataFromDB(String collectionName) {

        try {
            // Get a collection
            MongoCollection<Document> collection = database.getCollection(collectionName);

            FindIterable<Document> data = collection.find();
            return data;
        } catch (Exception e) {
            // TODO: handle exception
            System.out.println("Failed to fetch data");
            return null;

        }

    }
}
